### PR TITLE
[JENKINS-72112] Allow merge strategy inheritance in PodTemplates

### PR DESCRIPTION
**Pull request by vivienfricadelamadeus**
https://issues.jenkins.io/browse/JENKINS-72112

### Testing done

```[tasklist]
### Submitter checklist
- [ ] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [ ] Ensure that the pull request title represents the desired changelog entry
- [ ] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue
```

### CODE REVIEW

File: src/main/resources/org/csanchez/jenkins/plugins/kubernetes/PodTemplate/help-inheritYamlMergeStrategy.html [hunk 1]
```diff
@@ -0,0 +1 @@
+False by default
```
Comment (Vlatombe) [inline src/main/resources/org/csanchez/jenkins/plugins/kubernetes/PodTemplate/help-inheritYamlMergeStrategy.html:1]: Can you be more explicit on what this checkbox is doing?

File: src/main/resources/org/csanchez/jenkins/plugins/kubernetes/PodTemplate/config.jelly [hunk 2]
```diff
@@ -73,6 +73,10 @@
     </f:entry>
 
     <f:dropdownDescriptorSelector title="${%Yaml merge strategy}" field="yamlMergeStrategy" default="${descriptor.defaultYamlMergeStrategy}" />
+    
+    <f:entry field="inheritYamlMergeStrategy" title="${%Inherit yaml merge strategy}" >
+      <f:checkbox default="false"/>
```
Comment (Vlatombe) [inline src/main/resources/org/csanchez/jenkins/plugins/kubernetes/PodTemplate/config.jelly:78]: Already defaults to false so this is superfluous.
```suggestion
      <f:checkbox/>
```

**Review by Vlatombe — COMMENTED** [review]
Looks promising

File: src/main/resources/org/csanchez/jenkins/plugins/kubernetes/PodTemplate/help-inheritYamlMergeStrategy.html (hunk 1 above)
Comment (vivienfricadelamadeus) [reply src/main/resources/org/csanchez/jenkins/plugins/kubernetes/PodTemplate/help-inheritYamlMergeStrategy.html:1]: Yes, forgot to push last version :)

File: src/main/resources/org/csanchez/jenkins/plugins/kubernetes/PodTemplate/config.jelly (hunk 2 above)
Comment (vivienfricadelamadeus) [reply src/main/resources/org/csanchez/jenkins/plugins/kubernetes/PodTemplate/config.jelly:78]: Will do

**Comment by igor-nikiforov** [comment]
We are faced with this issue too.

@vivienfricadelamadeus are you planning to continue work on this PR? Thanks.

**Comment by vivienfricadelamadeus** [comment]
@igor-nikiforov Actually we found a workaround on our side (not directly linked to this feature), so I left the PR as it is for now. Don't hesitate to fork it

**Comment by Vlatombe** [comment]
Superceded by #1513

**Pull request closed without merging**